File: python/네트워크.py
from collections import deque


def solution(n, computers):
    answer = 0

    size = len(computers)

    check = [False for _ in range(size)]
    q = deque([])

    for i in range(size):
        if not check[i]:
            check[i] = True
            answer += 1
            q.append(i)

            while q:
                cur = q.popleft()

                for j in range(size):
                    if computers[cur][j] == 1 and not check[j]:
                        check[j] = True
                        q.append(j)

    return answer

File: js/네트워크.js
function solution(n, computers) {
    var answer = 0;

    var q = [];
    var check = new Array(n).fill(false);

    var cur;
    for (var idx = 0; idx < n; idx++) {
        if (check[idx] === false) {
            check[idx] = true;
            answer += 1;
            q.push(idx);
        }
        while (q.length > 0) {
            cur = q.shift();
            for (var i = 0; i < n; i++) {
                if (computers[cur][i] === 1 && check[i] === false) {
                    check[i] = true;
                    q.push(i);
                }
            }
        }
    }

    return answer;
}
